Sources/Overload/OvCore/include/OvCore/Global/ServiceLocator.h

```diff
@@ -6,15 +6,48 @@
 
 #pragma once
 
-#include <any>
-#include <unordered_map>
-
 #include "OvCore/API/Export.h"
 
 #define OVSERVICE(Type) OvCore::Global::ServiceLocator::Get<Type>()
 
 namespace OvCore::Global
 {
+	namespace Details
+	{
+		/**
+		 * Helper for service locating provides:
+		 *     1. Real O(1) access time
+		 *     2. Type-safe
+		 *     3. No RTTI required for service identification
+		 *     4. No key-value container required for service storage
+		 *     5. No type casting required for service retrieving
+		 */
+		template<typename T>
+		class Locator
+		{
+		public:
+			/**
+			 * Set new instance of service
+			 * @param p_service
+			 */
+			static void Set(T& p_service)
+			{
+				_SERVICE = &p_service;
```

Comment:
I would add an assert here (Using `OVASSERT`) that make sure `_SERVICE` is `nullptr` (We do not want to provide a service twice).